packages/plugins/jwt/src/plugin.ts

```diff
@@ -0,0 +1,242 @@
+import { createGraphQLError, type Plugin, type YogaLogger } from 'graphql-yoga';
+import jsonwebtoken, { type Jwt, type JwtPayload, type VerifyOptions } from 'jsonwebtoken';
+import { JwksClient } from 'jwks-rsa';
+import { type OnRequestParseEventPayload } from 'packages/graphql-yoga/src/plugins/types.js';
+import { normalizeConfig, type JwksSource, type JwtPluginOptions } from './config.js';
+import '@whatwg-node/server-plugin-cookies';
+import { GraphQLError } from 'graphql';
+
+export function useJWT(options: JwtPluginOptions): Plugin {
+  let logger: YogaLogger;
+  const normalizedOptions = normalizeConfig(options);
+  const payloadByRequest = new WeakMap<Request, JwtPayload | string>();
+  const signingKeyProviders = normalizedOptions.singingKeyProviders.map(provider =>
+    buildProvider(provider),
+  );
+  const lookupToken = async (payload: OnRequestParseEventPayload<object>) => {
+    for (const location of normalizedOptions.tokenLookupLocations) {
+      switch (location.type) {
+        case 'header': {
+          const header = payload.request.headers.get(location.name);
+
+          if (header == null) {
+            continue;
+          }
+
+          if (!location.prefix) {
+            return header;
+          }
+
+          const [prefix, token] = header.split(' ').map(s => s.trim());
+
+          if (prefix !== location.prefix) {
+            throw badRequestError(`Authentication token prefix does not match expected value.`);
+          }
```

Comment:
This means you can only have a `Prefix token` format ? Perhaps we don't need to assume that the prefix is seperated from the token with a space (and doesn't contain itself a space) ?

```suggestion
          const [prefix, token] = header.split(location.pefix).map(s => s.trim());

          if (prefix !== '') {
            throw badRequestError(`Authentication token prefix does not match expected value.`);
          }
```

Reply:
No, you can use `prefix: undefined` and it will not check the prefix. 
If a prefix is set, it needs to be strictly checked.